manifests/init.pp

```diff
@@ -102,6 +102,9 @@
   }
 
   $nss_db_dir   = "${pki_dir}/nssdb"
+  $nss_db_password_file = 'nss_db_password-file'
+  $nss_db_password_path = "${pki_dir}/${nss_db_password_file}"
+  $nss_db_password = extlib::cache_data('foreman_cache_data', $nss_db_password_file, extlib::random_password(32))
```

Comment:
We could use `basename` (https://github.com/puppetlabs/puppetlabs-stdlib/blob/main/REFERENCE.md#basename) here passing in the current `$nss_db_password_file`

Reply:
@ehelms do you feel strongly that this is the better alternative vs. making the simple change in puppet-katello I suggested on the other comment?

Reply:
This captures my thoughts -- https://github.com/theforeman/puppet-certs/pull/344#discussion_r655618132

If you want to make this naming change, I would prefer if you split it out to its own issue, we merge that change and the puppet-katello change then layer this on top. Or vice versa.

Reply:
The `foreman_cache_data` cache namespace feels odd to me. If you read https://github.com/voxpupuli/puppet-extlib/blob/master/lib/puppet/functions/extlib/cache_data.rb you'll see it ends up in `$vardir/foreman_cache_data/nss_db_password-file.yaml`. On the other hand, we don't want to pollute `$vardir` with many directories. However, I think this would at least be clearer:
```suggestion
  $nss_db_password = extlib::cache_data('foreman_cache_data', 'certs-nss-db-password', extlib::random_password(32))
```

Reply:
Thanks, @ekohl . That's a good point.

Looking around a bit, in pulpcore we use the `pulpcore_cache_data` cache namespace. But in puppet-candlepin, it's also `foreman_cache_data` which is a bit awkward: https://github.com/theforeman/puppet-candlepin/blob/b3b234376b48d00db22f8cb91913aa3e12bc964d/manifests/params.pp#L6

As it's currently written, changing the name of the password file which would necessarily mean changing the password itself. With your suggestion we could rename the file independently of changing the password, although I'm not sure if there is ever a real reason to do so -- I think it would still cause a refresh of the entire resource either way.

I would say, if we want to make that change, then let's remember to address it here as well: https://github.com/theforeman/puppet-certs/blob/fe831c38c8488be70ca28bb41dfb24854d94a8a6/manifests/candlepin.pp#L71-L72 .